device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -146,19 +178,27 @@ public void open() throws IOException
      * <p>
      * Once closed, this client can be re-opened. It will preserve all previously registered device clients.
      * <p>
-     * @throws IOException If any exception occurs while closing the connection.
+     * @throws MultiplexingClientException If any IO errors occur when fulfilling this request.
      */
-    public void close() throws IOException
+    public void close() throws MultiplexingClientException
     {
         synchronized (this.operationLock)
         {
             log.info("Closing multiplexing client");
-            for (DeviceClient deviceClient : this.multiplexedDeviceClients.values())
+
+            try
             {
-                deviceClient.closeFileUpload();
-            }
+                for (DeviceClient deviceClient : this.multiplexedDeviceClients.values())
+                {
+                    deviceClient.closeFileUpload();
+                }
 
-            this.deviceIO.multiplexClose();
+                this.deviceIO.closeWithoutWrappingException();
+            }
+            catch (TransportException | IOException e)
```

Comment:
Q - if we get an IOException here, should we extract the transport exception from the IOException and throw that within the new MultiplexingClientException?

Reply:
To avoid the actual exception info being 2 levels lower

Reply:
The IOException caught here is only thrown from the closing of the FileUpload code. That code doesn't have nested TransportExceptions, so it should be fine to not try to un-nest this particular exception